core/src/stores/block_information_store.rs

```diff
@@ -164,4 +166,14 @@ impl BlockInformationStore {
             None => (false, latest_block.slot),
         }
     }
+
+    pub fn get_block_info_by_slot(&self, slot: u64) -> Option<BlockInformation> {
```

Comment:
`blocks: Arc<DashMap<String, BlockInformation>>`
^^please provide a comment what "String" is (or better, use a custom type)